### PR TITLE
Hotreload Fix

### PR DESCRIPTION
**Pull request by voidentente**
A quick fix I threw together. Sorry if it isn't very clean, but hotloading local files should work again now.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 mod web_asset_io;
 mod web_asset_plugin;
+mod web_filesystem_watcher;
 
 pub use web_asset_io::WebAssetIo;
 pub use web_asset_plugin::WebAssetPlugin;
+pub use web_filesystem_watcher::FilesystemWatcher;
```
Comment (johanhelsing) [inline src/lib.rs]: Does this need to be public?

File: src/lib.rs [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 mod web_asset_io;
 mod web_asset_plugin;
+mod web_filesystem_watcher;
```
Comment (johanhelsing) [inline src/lib.rs]: i think the web prefix here is slightly confusing. Maybe `local_filesystem_watcher` or just `filesystem_watcher`?

File: src/web_filesystem_watcher.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                sender.send(res).expect("Watch event send failure.");
+            },
+            Config::default(),
+        )
+        .expect("Failed to create filesystem watcher.");
+        FilesystemWatcher { watcher, receiver }
+    }
+}
+
+impl FilesystemWatcher {
```
Comment (johanhelsing) [inline src/web_filesystem_watcher.rs]: Is this implementation based on the regular bevy filsystem watcher?

File: src/web_asset_io.rs [hunk 4]
```diff
@@ -68,19 +72,36 @@ impl AssetIo for WebAssetIo {
         self.default_io.read_directory(path)
     }
 
-    fn watch_path_for_changes(&self, path: &Path) -> Result<(), AssetIoError> {
-        if is_http(path) {
+    fn watch_path_for_changes(&self, _path: &Path) -> Result<(), AssetIoError> {
+        if is_http(_path) {
```
Comment (johanhelsing) [inline src/web_asset_io.rs]: Don't prefix arguments with _ unless they're unused

File: src/web_filesystem_watcher.rs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+pub fn filesystem_watcher_system(asset_server: Res<AssetServer>) {
+    let mut changed = HashSet::default();
+
+    let asset_io = if let Some(asset_io) = asset_server.asset_io().downcast_ref::<WebAssetIo>() {
+        asset_io
+    } else {
+        return;
+    };
+
+    let result = asset_io.filesystem_watcher.read();
```
Comment (johanhelsing) [inline src/web_filesystem_watcher.rs]: Maybe these (option, result) variables could be given more descriptive names? Or inline them?

**Review by johanhelsing — COMMENTED** [review]
Thanks for the PR!

I did a quick review, though i don't fully understand the actual watcher yet. Will have a closer look when i have more time :)

File: src/web_filesystem_watcher.rs (hunk 3 above)
Comment (voidentente) [reply src/web_filesystem_watcher.rs]: The FilesystemWatcher struct was completely copied from the default implementation, which is not public.

File: src/web_asset_io.rs (hunk 4 above)
Comment (voidentente) [reply src/web_asset_io.rs]: This is directly copied from the FileAssetIo implementation; I thought this was weird, too, I don't know why I kept it in.

File: src/web_filesystem_watcher.rs (hunk 3 above)
Comment (johanhelsing) [reply src/web_filesystem_watcher.rs]: Okay, makes sense :) Maybe just add a comment with a link to the bevy implementation, so we know where to look when updating bevy version?

**Comment by johanhelsing** [comment]
Nice work, thanks a lot :)

**Pull request merged**